onediff_diffusers_extensions/diffusers_extensions/deep_cache/pipeline_stable_video_diffusion.py

```diff
@@ -0,0 +1,375 @@
+from diffusers.pipelines.pipeline_utils import DiffusionPipeline
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+from diffusers import StableVideoDiffusionPipeline as OrgStableVideoDiffusionPipeline
+from diffusers.pipelines.stable_video_diffusion import StableVideoDiffusionPipelineOutput
+
+
+from .models.unet_spatio_temporal_condition import UNetSpatioTemporalConditionModel
+from .models.fast_unet_spatio_temporal_condition import FastUNetSpatioTemporalConditionModel
+
+
+from .models.pipeline_utils import enable_deep_cache_pipeline
+
+
+enable_deep_cache_pipeline()
+
+def _append_dims(x, target_dims):
+    """Appends dimensions to the end of a tensor until it has target_dims dimensions."""
+    dims_to_append = target_dims - x.ndim
+    if dims_to_append < 0:
+        raise ValueError(f"input has {x.ndim} dims but target_dims is {target_dims}, which is less")
+    return x[(...,) + (None,) * dims_to_append]
+
+
+def tensor2vid(video: torch.Tensor, processor, output_type="np"):
+    # Based on:
+    # https://github.com/modelscope/modelscope/blob/1509fdb973e5871f37148a4b5e5964cafd43e64d/modelscope/pipelines/multi_modal/text_to_video_synthesis_pipeline.py#L78
+
+    batch_size, channels, num_frames, height, width = video.shape
+    outputs = []
+    for batch_idx in range(batch_size):
+        batch_vid = video[batch_idx].permute(1, 0, 2, 3)
+        batch_output = processor.postprocess(batch_vid, output_type)
+
+        outputs.append(batch_output)
+
+    return outputs
+
+
+class StableVideoDiffusionPipeline(OrgStableVideoDiffusionPipeline):
+    r"""
+    Pipeline to generate video from an input image using Stable Video Diffusion.
+
+    This model inherits from [`DiffusionPipeline`]. Check the superclass documentation for the generic methods
+    implemented for all pipelines (downloading, saving, running on a particular device, etc.).
+
+    Args:
+        vae ([`AutoencoderKL`]):
+            Variational Auto-Encoder (VAE) model to encode and decode images to and from latent representations.
+        image_encoder ([`~transformers.CLIPVisionModelWithProjection`]):
+            Frozen CLIP image-encoder ([laion/CLIP-ViT-H-14-laion2B-s32B-b79K](https://huggingface.co/laion/CLIP-ViT-H-14-laion2B-s32B-b79K)).
+        unet ([`UNetSpatioTemporalConditionModel`]):cache_interval=5, cache_branch=0,
+            A `UNetSpatioTemporalConditionModel` to denoise the encoded image latents.
+        scheduler ([`EulerDiscreteScheduler`]):
+            A scheduler to be used in combination with `unet` to denoise the encoded image latents.
+        feature_extractor ([`~transformers.CLIPImageProcessor`]):
+            A `CLIPImageProcessor` to extract features from generated images.
+    """
+
+    model_cpu_offload_seq = "image_encoder->unet->vae"
+    _callback_tensor_inputs = ["latents"]
+
+    def __init__(
+        self,
+        vae: AutoencoderKLTemporalDecoder,
+        image_encoder: CLIPVisionModelWithProjection,
```

Comment:
Seems like this component fails to get saved, FYI:
```
2024-01-12 01:11:46.164 [stderr   ]   File "/root/.cache/isolate/virtualenv/5371bcd257691477725a108bc5f62c9ccb52f53db92ecf3cddd4ceb4111f24d0/lib/python3.11/site-packages/onediff/infer_compiler/with_oneflow_compile.py", line 406, in save_graph
2024-01-12 01:11:46.164 [stderr   ]     flow.save(state_dict, file_path)
2024-01-12 01:11:46.164 [stderr   ]   File "/root/.cache/isolate/virtualenv/5371bcd257691477725a108bc5f62c9ccb52f53db92ecf3cddd4ceb4111f24d0/lib/python3.11/site-packages/oneflow/framework/check_point_v2.py", line 720, in save
2024-01-12 01:11:46.164 [stderr   ]     pickled_bytes = pickle.dumps(obj)
2024-01-12 01:11:46.164 [stderr   ]                     ^^^^^^^^^^^^^^^^^
2024-01-12 01:11:46.164 [stderr   ] _pickle.PicklingError: Can't pickle <class 'transformers.models.clip.modeling_clip.CLIPVisionModelOutput'>: it's not the same object as transformers.models.clip.modeling_clip.CLIPVisionModelOutput
```

Reply:
Compilation of image_encoder can be skipped, we have not tested its graph save.